### PR TITLE
VerifierState deprecated and warning about equals/hashcode removed.

### PR DESCRIPTION
**Pull request by avpotapov00**
```requireStateEquivalenceImplCheck``` option was remove and ```VerifierState``` was marked deprecated.
Annoying warnings about ```equals/hashcode``` removed were removed.

See #135 for details.

### CODE REVIEW

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt [hunk 1]
```diff
@@ -177,27 +180,20 @@ class LinChecker (private val testClass: Class<*>, options: Options<*, *>?) {
...
-            if (!stateEquivalenceCorrect) {
-                if (requireStateEquivalenceImplCheck) {
-                    val errorMessage = StringBuilder().appendStateEquivalenceViolationMessage(sequentialSpecification).toString()
-                    error(errorMessage)
-                } else {
-                    reporter.logStateEquivalenceViolation(sequentialSpecification)
-                }
-            }
-        }
+    private val ExecutionScenario.hasSuspendableActorsInInitPart
```
Comment (ndkoval) [inline src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt:183]: Please revert everything except for the "createVerifier" function

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt [hunk 2]
```diff
@@ -64,31 +64,30 @@ class LinChecker (private val testClass: Class<*>, options: Options<*, *>?) {
...
-                verifier = createVerifier(checkStateEquivalence = false)
+                verifier = createVerifier()
             val scenario = exGen.nextExecution()
             scenario.validate()
             reporter.logIteration(i + 1 + customScenarios.size, iterations, scenario)
             val failure = scenario.run(this, verifier)
             if (failure != null) {
                 val minimizedFailedIteration = if (!minimizeFailedScenario) failure
-                                               else failure.minimize(this)
+                else failure.minimize(this)
```
Comment (ndkoval) [inline src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt:90]: Please revert or change to one-line statement

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt [hunk 3]
```diff
@@ -39,7 +39,7 @@ class LinChecker (private val testClass: Class<*>, options: Options<*, *>?) {
         val logLevel = options?.logLevel ?: testClass.getAnnotation(LogLevel::class.java)?.value ?: DEFAULT_LOG_LEVEL
         reporter = Reporter(logLevel)
         testConfigurations = if (options != null) listOf(options.createTestConfigurations(testClass))
-                             else createFromTestClassAnnotations(testClass)
+        else createFromTestClassAnnotations(testClass)
```
Comment (ndkoval) [inline src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt:42]: Please revert

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/verifier/Verifier.java [hunk 4]
```diff
@@ -33,17 +33,11 @@
  * All implementations should have {@code (Class<?> sequentialSpecification)} constructor,
  * which takes the scenario to be tested and the correct sequential implementation of the testing data structure.
  */
+@FunctionalInterface
```
Comment (ndkoval) [inline src/jvm/main/org/jetbrains/kotlinx/lincheck/verifier/Verifier.java:36]: I don't think you are going to it as a functional interface :)

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/verifier/VerifierState.kt [hunk 5]
```diff
@@ -27,6 +27,7 @@ package org.jetbrains.kotlinx.lincheck.verifier
  *
  * [VerifierState] lazily counts the test instance state, caches it, and uses in `equals` and `hashCode` methods.
  */
+@Deprecated("Doesn't always improve performance of verification", level = DeprecationLevel.WARNING)
```
Comment (ndkoval) [inline src/jvm/main/org/jetbrains/kotlinx/lincheck/verifier/VerifierState.kt:30]: Please use both Java's and Kotlin's `@Deprecation` annotations.

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/Options.kt [hunk 6]
```diff
@@ -112,15 +111,6 @@ abstract class Options<OPT : Options<OPT, CTEST>, CTEST : CTestConfiguration> {
         this.verifier = verifier
     }
 
-    /**
-     * Require correctness check of test instance state equivalency relation defined by the user.
-     * It checks whether two new instances of a test class are equal.
-     * If the check fails [[IllegalStateException]] is thrown.
-     */
-    fun requireStateEquivalenceImplCheck(require: Boolean): OPT = applyAndCast {
```
Comment (ndkoval) [inline src/jvm/main/org/jetbrains/kotlinx/lincheck/Options.kt:120]: Please do not remove it, deprecate instead

**Review by ndkoval — CHANGES_REQUESTED** [review]

**Comment by ndkoval** [comment]
Please also resolve the conflicts

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt (hunk 1 above)
Comment (avpotapov00) [reply src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt:183]: Done

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt (hunk 2 above)
Comment (avpotapov00) [reply src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt:90]: Done

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt (hunk 3 above)
Comment (avpotapov00) [reply src/jvm/main/org/jetbrains/kotlinx/lincheck/LinChecker.kt:42]: Done

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/Options.kt (hunk 6 above)
Comment (avpotapov00) [reply src/jvm/main/org/jetbrains/kotlinx/lincheck/Options.kt:120]: Done

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/verifier/VerifierState.kt (hunk 5 above)
Comment (avpotapov00) [reply src/jvm/main/org/jetbrains/kotlinx/lincheck/verifier/VerifierState.kt:30]: Done

**Review by ndkoval — CHANGES_REQUESTED** [review]
Please rebase on `develop`

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/modelchecking/ModelCheckingCTest.java [hunk 7]
```diff
@@ -113,10 +113,13 @@
     int invocationsPerIteration() default DEFAULT_INVOCATIONS;
 
     /**
+     * Does nothing, states equivalence does not always improve performance of verification.
+     * <p>
      * Require correctness check of test instance state equivalency relation, which is defined by the user.
      * Essentially, it checks whether two new instances of the test class are equal.
      * If the check fails, an {@link IllegalStateException} is thrown.
      */
+    @Deprecated
```
Comment (ndkoval) [inline src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/modelchecking/ModelCheckingCTest.java:122]: Kotlin's `@Deprecated`

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/verifier/VerifierState.kt [hunk 8]
```diff
@@ -27,6 +27,8 @@ package org.jetbrains.kotlinx.lincheck.verifier
...
+@java.lang.Deprecated
```
Comment (ndkoval) [inline src/jvm/main/org/jetbrains/kotlinx/lincheck/verifier/VerifierState.kt]: Let's use only Kotlin's `@Deprecated` :)

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/Options.kt [hunk 9]
```diff
@@ -113,13 +112,15 @@ abstract class Options<OPT : Options<OPT, CTEST>, CTEST : CTestConfiguration> {
...
+     *
+     * Required correctness check of test instance state equivalency relation defined by the user.
+     * It checked whether two new instances of a test class are equal.
+     * If the check failed [[IllegalStateException]] was thrown.
      */
-    fun requireStateEquivalenceImplCheck(require: Boolean): OPT = applyAndCast {
-        requireStateEquivalenceImplementationCheck = require
-    }
+    @Deprecated("Does nothing, because equals/hashcode don't always improve performance of verification")
+    @java.lang.Deprecated
```
[the remark this answers is not in the document]
Comment (avpotapov00) [reply src/jvm/main/org/jetbrains/kotlinx/lincheck/Options.kt]: Done

**Pull request merged**